### PR TITLE
adds jwk alg matcher

### PR DESCRIPTION
**Pull request by kevprice83**
This PR adds a function to match the algorithm of the public key retrieved from the `jwks_uri` against the `jwt.header.alg` field. Currently we only match the JWT's `alg` against a whitelist which is populated based on the response from the discovery endpoint but this further check is required to ensure we do the verification correctly.

Fixes [THREESCALE-8249](https://issues.redhat.com/browse/THREESCALE-8249)

### CODE REVIEW

**Review by eloycoto — APPROVED** [review]
woah, good one.

**Comment by gsaslis** [comment]
@eloycoto I am wondering if we can have this PR merged, but I see a couple of broken CI checks. 

On the one hand, I see the `prove` job has been failing ... almost consistently in - at least - a few other PRs I checked. As such, I am inclined to ignore it. 

On the other hand, I am not sure about the `busted` job failure? Should that failure block the merging of this PR, do you think?

Thanks in advance!

**Comment by eloycoto** [comment]
Both should work, someone should have a look at that, both are important.

**Comment by kevprice83** [comment]
@gsaslis before merging I need to write the integration & unit tests. At the moment it's not urgent as we are planning to included this in the 2.12 release but I have it as a top priority for next week!

File: gateway/.s2i/bin/assemble
Comment (samugi) [inline gateway/.s2i/bin/assemble:8]: I believe this was maybe included by mistake in this PR?

**Comment by samugi** [comment]
assuming the circleci failures are all due to external factors: lgtm

**Review by samugi — CHANGES_REQUESTED** [review]
assuming the circleci failures are all due to external factors: lgtm

**Comment by eguzki** [comment]
LGTM

I share the [comment](https://github.com/3scale/APIcast/pull/1329#discussion_r838632920) for the change in `gateway/.s2i/bin/assemble`

Other than that, is should be rebased on top of master. 

Great work

File: gateway/.s2i/bin/assemble
Comment (kevprice83) [reply gateway/.s2i/bin/assemble:8]: well it's not a big deal but yeah cleaner to remove it I guess

Comment (kevprice83) [reply gateway/.s2i/bin/assemble]: Done!

**Review by samugi — APPROVED** [review]
lgtm

**Pull request merged**
